crates/indexer/src/models/property_map.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+use serde_json::Result;
+use serde_json::Value;
+use std::collections::HashMap;
+
+use crate::util;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct PropertyValue {
+    value: String,
+    typ: String,
+}
+
+pub fn create_property_value(typ: String, value: String) -> Result<PropertyValue> {
+    Ok(PropertyValue {
+        value: util::convert_bcs_hex(typ.clone(), value.clone()).unwrap_or(value),
+        typ,
+    })
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct PropertyMap {
+    data: HashMap<String, PropertyValue>,
+}
+
+impl PropertyMap {
+    /// Deserializes PropertyValue from bcs encoded json
+    pub fn from_bsc_encode_str(val: Value) -> Option<Value> {
+        let mut pm = PropertyMap {
+            data: HashMap::new(),
+        };
+        let records: &Vec<Value> = val.get("map")?.get("data")?.as_array()?;
```

Comment:
you mean {data: {blob}} instead of {blob}? I don't like it though, a property map is just a map right? Why does downstream have to know that it's a property map?

Reply:
ooh constants. nvm idk why i understood it at something else.